scene/3d/cpu_particles_3d.cpp

```diff
@@ -1156,7 +1156,7 @@ void CPUParticles3D::_particles_process(double p_delta) {
 
 		if (particle_flags[PARTICLE_FLAG_DISABLE_Z]) {
 			if (particle_flags[PARTICLE_FLAG_ALIGN_Y_TO_VELOCITY]) {
-				if (p.velocity.length() > 0.0) {
+				if (p.velocity.length_squared() > 0.0) {
```

Comment:
Pre-existing problem, but all the changed comparisons in this file should be given an `f` suffix.

```suggestion
				if (p.velocity.length_squared() > 0.0f) {
```